dateparser/date.py

```diff
@@ -254,6 +252,40 @@ def _is_valid_date_obj(self, date_obj):
         return True
 
 
+class DateData:
+    """
+    Class that represents the parsed data with useful information.
+    It can be accessed like a dict object.
+    """
+
+    def __init__(self, date_obj=None, period=None, locale=None, is_relative=None):
```

Comment:
I think it makes sense to force keyword args for this class.

```suggestion
    def __init__(self, *, date_obj=None, period=None, locale=None, is_relative=None):
```

Reply:
I don't understand this. Could you elaborate on it a little? Why should we allow other fields than the ones defined? for avoiding issues with future deprecations? :thinking:

Reply:
> Why should we allow other fields than the ones defined?

That’s not what the `*` does, it’s not the same as `*<variable name>` (e.g. `*args`). A single `*` as a parameter simply causes all parameters after it to require to be passed by name. See https://python-3-for-scientists.readthedocs.io/en/latest/python3_advanced.html#keyword-only-arguments

So, before this change, you can create an instance of `DateData` class 2 different ways:

```python
DateData(date_object, 'day', 'gl_ES', False)
DateData(date_object=date_object, period='day', locale='gl_ES', is_relative=True)
```

With my suggestion, only the latter works.

With classes and functions where positional parameter are not obvious, keyword-only arguments are a way to force calling code to remain readable.

I don’t have a strong opinion on this, though. Feel free to ignore and mark as resolved.